arch/arm/core/aarch32/cortex_m/Kconfig

```diff
@@ -23,6 +23,14 @@ config CPU_CORTEX_M0PLUS
 	help
 	  This option signifies the use of a Cortex-M0+ CPU
 
+config CPU_CORTEX_M1
+	bool
+	select CPU_CORTEX_M
+	select ARMV6_M_ARMV8_M_BASELINE
+	select CPU_CORTEX_M_HAS_SYSTICK
```

Comment:
The SysTick timer is enabled when selecting `OS Extensions` under the Cortex-M1 customisation options. These include extensions include both the SysTick, SVC, and the NVIC. I am not sure it is worth attempting to support a Cortex-M1 with `OS extensions` disabled.

Reply:
It does not matter, IMO. We cannot list the HAS_SYSTICK as a capability of Cortex-M1, since it is optional. Adding it will create confusion on which features are always-present, and it will require re-write as soon as there's an FPGA without the systick.

Reply:
I see your point. How do you propose we handle the NVIC, then? It too depends on `OS Extensions`.

Reply:
There is no Kconfig option related to NVIC, as far as I remember. So from Kconfig's perspective we are fine, for now. 
In Cortex-M, you can use `CONFIG_NUM_IRQS = 0`, to signify a CPU with no NVIC (so the core has only the ARM exceptions).

Reply:
Compiling with `CONFIG_NUM_IRQS=0` gives compilation errors in `isr_tables.c`. I suggest leaving it at 32 on the SoC level.

Reply:
OK, this should be fixed. In my opinion CONFIG_NUM_IRQS=0 should be a valid configuration. Maybe we should open a ticket.

Reply:
@ioannisg I have opened #27490 to track that issue.

Reply:
After discussing this on Slack, I have moved `CONFIG_CPU_CORTEX_M_HAS_SYSTICK=y` to the board level and removed `CONFIG_NUM_IRQS=32` default from the SoC level.